### PR TITLE
vcftools: add v0.1.16

### PR DESCRIPTION
**Pull request by snehring**
Version bump

### CODE REVIEW

File: var/spack/repos/builtin/packages/vcftools/package.py [hunk 1]
```diff
@@ -26,7 +27,15 @@ class Vcftools(AutotoolsPackage):
...
-        return ["--with-pmdir={0}".format(self.prefix.lib)]
+        ret = []
+        # between 0.1.16 and 14 the behavior of the configure script
+        # wrt the perl lib dir changed and it became relative to the
+        # install directory, if you specify the whole prefix in
+        # it now you end up with a nasty recreation of the
+        # prefix tree in self.prefix.
+        if self.spec.satisfies("@0.1.14"):
+            ret.append("--with-pmdir={0}".format(self.prefix.lib))
+        return ret
```
Comment (alecbcs) [inline var/spack/repos/builtin/packages/vcftools/package.py]:
```suggestion
        args = []
        # between 0.1.16 and 14 the behavior of the configure script
        # wrt the perl lib dir changed and it became relative to the
        # install directory, if you specify the whole prefix in
        # it now you end up with a nasty recreation of the
        # prefix tree in self.prefix.
        if self.spec.satisfies("@:0.1.14"):
            args.append(f"--with-pmdir={self.prefix.lib}")
        return args
```

In keeping with Spack conventions I'd name the variable here `args`. (Also format strings make everything simpler and we're moving to them more generally.)

**Review by alecbcs — CHANGES_REQUESTED** [review]
Quick formatting suggestion. Otherwise this PR looks good to me. Thanks @snehring!

File: var/spack/repos/builtin/packages/vcftools/package.py (hunk 1 above)
Comment (snehring) [reply var/spack/repos/builtin/packages/vcftools/package.py]: Agree with format strings, just held that here given prior convention.

**Review by alecbcs — APPROVED** [review]
Confirmed SHA. @snehring thanks for the quick turn around on those suggestions!

**Pull request merged**